rp2040-hal/src/spi.rs

```diff
@@ -16,7 +19,11 @@
 //! let sclk = pins.gpio2.into_function::<FunctionSpi>();
 //! let mosi = pins.gpio3.into_function::<FunctionSpi>();
 //!
-//! let spi = Spi::<_, _, _, 8>::new(peripherals.SPI0, (mosi, sclk)).init(&mut peripherals.RESETS, 125_000_000u32.Hz(), 16_000_000u32.Hz(), MODE_0);
+//! let spi_device = peripherals.SPI0;
+//! let spi_pin_layout = (mosi, sclk);
+//!
+//! let spi = Spi::new(spi_device, spi_pin_layout)
```

Comment:
This still fails to compile:
```
help: consider giving `spi` an explicit type, where the value of const parameter `DS` is specified
```
That's a little bit unfortunate, as the type annotation clutters the code. Any suggestion on how to express this in a more readable way would be welcome!
```suggestion
//! let spi = Spi::<_, _, _, 8>::new(spi_device, spi_pin_layout)
```

Reply:
I made https://github.com/fu5ha/rp-hal/pull/1 on top of this which is unfortunately a breaking change but would improve the type inference ergonomics for this. It's a bit unfortunate Rust can't infer the type default as-is, but yeah.